include/ibamr/EnthalpyHierarchyIntegrator.h

```diff
@@ -0,0 +1,249 @@
+     * \param from The value to copy to this object.
+     */
+    EnthalpyHierarchyIntegrator(const EnthalpyHierarchyIntegrator& from) = delete;
+
+    /*!
+     * \brief Assignment operator.
+     *
+     * \note This operator is not implemented and should not be used.
+     *
+     * \param that The value to assign to this object.
+     *
+     * \return A reference to this object.
+     */
+    EnthalpyHierarchyIntegrator& operator=(const EnthalpyHierarchyIntegrator& that) = delete;
+
+    /*!
+     * Read input values from a given database.
+     */
+    void getFromInput(SAMRAI::tbox::Pointer<SAMRAI::tbox::Database> input_db, bool is_from_restart);
+
+    /*!
+     * \brief Read object state from the restart file and initialize class data
+     * members.
+     */
+    void getFromRestart();
+
+    /*!
+     * \brief Compute enthalpy based on (nonlinear) h-T relation.
+     */
+    void
+    computeEnthalpyBasedOnTemperature(int h_idx, const int T_idx, const int rho_idx, const int lf_idx, const int H_idx);
+
+    /*!
+     * \brief Compute temperature based on (nonlinear) h-T relation.
+     */
+    void computeTemperatureBasedOnEnthalpy(int T_idx, const int h_idx, const int H_idx);
+
+    /*!
+     * \brief Update enthalpy using Taylor series \f$ h_{\rm new} = h_{\rm old} + dh/dt (\Delta T) \f$.
+     */
+    void updateEnthalpy(int h_idx, const int T_new_idx, const int T_pre_idx);
+
+    /*!
+     * \brief compute dh/dT based on temperature.
+     */
+    void computeEnthalpyDerivative(int dh_dT_data, const int T_idx, const int H_idx);
+
+    /*!
+     * \brief Compute liquid fraction.
+     */
+    void computeLiquidFraction(int lf_idx, const int h_idx, const int H_idx);
+
+    /*!
+     * Solver variables.
+     */
+    SAMRAI::tbox::Pointer<SAMRAI::pdat::CellVariable<NDIM, double> > d_h_var;
+    SAMRAI::tbox::Pointer<SAMRAI::pdat::SideVariable<NDIM, double> > d_grad_T_var;
+    SAMRAI::tbox::Pointer<SAMRAI::pdat::CellVariable<NDIM, double> > d_T_pre_var;
+    SAMRAI::tbox::Pointer<SAMRAI::pdat::CellVariable<NDIM, double> > d_dh_dT_var;
+
+    SAMRAI::solv::RobinBcCoefStrategy<NDIM>* d_h_bc_coef = nullptr;
+
+    /*!
+     * Patch data descriptor indices for all "state" variables managed by the
+     * integrator.
+     *
+     * State variables have three contexts: current, scratch, and new.
+     */
+    int d_h_scratch_idx = IBTK::invalid_index, d_h_current_idx = IBTK::invalid_index, d_h_new_idx = IBTK::invalid_index;
+
+    /*!
+     * Patch data descriptor indices for all "scratch" variables managed by the
+     * integrator.
+     *
+     * Scratch variables have only one context: scratch.
+     */
+    int d_grad_T_idx = IBTK::invalid_index;
+    int d_T_pre_idx = IBTK::invalid_index;
+    int d_dh_dT_scratch_idx = IBTK::invalid_index;
+
+    /*!
+     * Energy equation parameters.
+     */
+    double d_liquidus_temperature, d_solidus_temperature, d_reference_temperature, d_Cp_liquid, d_Cp_solid, d_Cp_gas,
+        d_Cp_mushy;
```

Comment:
Could you use `heat_capacity` instead of `Cp` here and mention in a comment that it is for (IIRC) constant pressure?